src/xrdok.js

```diff
@@ -681,3 +681,280 @@ const XRiconchevron = 'xr-icon-chevron';
+      linePath.setAttribute('line',
+        `start: ${-data.secondaryValuesBase} ${prim} ${-j};
+         end: ${data.secondaryValuesBase} ${prim} ${-j};
+         color: white;
+        `
+      );
+    }
+
+    // create primary material
+    planeGraphPrimaryMat.side = THREE.DoubleSide;
+    planeGraphPrimaryMat.transparent = true;
+    planeGraphPrimaryMat.opacity = 1;
+    planeGraphPrimaryMat.color.set( data.colorMin );
+
+    // create secondary material
+    planeGraphSecondaryMat.side = THREE.DoubleSide;
+    planeGraphSecondaryMat.transparent = true;
+    planeGraphSecondaryMat.opacity = 1;
+    planeGraphSecondaryMat.color.set( data.colorMax );
+
+    // create meshes
+    var planeGraphPrimaryMesh = new THREE.Mesh(planeGraphPrimaryGeo, planeGraphPrimaryMat);
+    planeGraphPrimaryEl.setObject3D('xr-primary-plane-graph', planeGraphPrimaryMesh);
+    var planeGraphSecondaryMesh = new THREE.Mesh(planeGraphSecondaryGeo, planeGraphSecondaryMat);
+    planeGraphSecondaryEl.setObject3D('xr-secondary-plane-graph', planeGraphSecondaryMesh);
+
+    planeGraphEl.appendChild(planeGraphPrimaryEl);
+    planeGraphEl.appendChild(planeGraphSecondaryEl);
+
+    planeGraphEl.setAttribute('visible', data.showGraph);
+    return planeGraphEl;
+  }
+
+  function createData(data, el) {
+    // parse CSV, if available
+    if (data.src) {
+      Papa.parse(data.src, {  // eslint-disable-line no-undef
+        download: true,
+        complete: function(res) {
+          data.descriptors = [];
+          data.primaryValues = [];
+          data.secondaryValues = [];
+          for (var i = 0; i < res.data.length; i++) {
+            data.descriptors.push(res.data[i][0]);
+            data.primaryValues.push(parseFloat(res.data[i][1]));
+            data.secondaryValues.push(parseInt(res.data[i][2]));
+          }
+
+          el.appendChild(createGeometryLine(data));
+          el.appendChild(createGeometryPlane(data));
+          el.appendChild(createLegend(data));
+        }
+      });
+    }
+  }
+
+  AFRAME.registerComponent(XRplanegraph, {
+
+    schema: {
+      color: { type: 'color', default: '#ccfccf'},
+      colorMax: { type: 'color', default: '#fccfcc'},
+      colorMin: { type: 'color', default: '#ccfccf'},
+      descriptors: {type: 'array', default: [] },
+      primaryValues: { type: 'array', default: [] },
+      primaryValuesBase: { type: 'number', default: 0},
+      secondaryValues: { type: 'array', default: [] },
+      secondaryValuesBase: { type: 'number', default: 1 },
+      showGraph: { type: 'boolean', default: true},
+      showLegend: { type: 'boolean', default: true},
+      showLine: { type: 'boolean', default: true},
+      src: { type: 'string', default: './models/lufttemperatur_bbb_sortiert.csv'},
+    },
+
+    init: function() {
+      // data and element
+      var data = this.data;
+      var el = this.el;
+
+      createData(data, el);
+
+      this.lineEl = document.getElementById('line-graph-parent');
```

Comment:
took me a while to understand what you are doing here..:)

it would be much easier to understand if you just return these values from `createData` function like you did in all the other create functions.

Also I think there is a bug here. Since `Papa.parse` `complete` callback can be called after you make the assignments here. Meaning you could get null values in your members as the referenced dom elements are just not there yet.
You need to assign these members after `Papa.parse` completes.